prestoadmin/server.py

```diff
@@ -82,11 +86,191 @@ def new_sysnode_processor(node_info_rows):
 CONNECTOR_INFO_SQL = 'select catalog_name from system.metadata.catalogs'
 _LOGGER = logging.getLogger(__name__)
 
+DOWNLOAD_DIRECTORY = main_dir
+DEFAULT_RPM_NAME = 'presto-server-rpm.rpm'
+
+
+def _find_or_download_most_recent_presto_rpm():
+    newest_rpm_release = 'https://repository.sonatype.org/service/local/artifact/maven' \
+                         '/content?r=central-proxy&g=com.facebook.presto' \
+                         '&a=presto-server-rpm&e=rpm&v=RELEASE'
+    return _find_or_download_rpm(newest_rpm_release)
+
+
+def _check_if_valid_version(rpm_version):
+    return re.match('^[0-9]+(\.[0-9]+){1,2}[tf]?$', rpm_version)
+
+
+def _check_good_response_status(response_status):
+    return response_status == 200
+
+
+def _get_content_length(url_response):
+    try:
+        headers = url_response.info()
+        return int(headers['Content-Length'])
+    except (KeyError, ValueError):
+        return None
+
+
+def _get_download_file_name(url_response, version=None):
+    try:
+        headers = url_response.info()
+        content_disposition = headers['Content-Disposition']
+        values, params = cgi.parse_header(content_disposition)
+        return params['filename']
+    except KeyError:
```

Comment:
HTTP makes no guarantee that the response will contain the `Content-Disposition` or the `Content-Length` headers. This catches the cases where the responses do not have those headers, so the program can continue.